ga4gh/backend.py

```diff
@@ -209,6 +214,83 @@ def getReadGroupSets(self):
         """
         return list(self._readGroupSetIdMap.values())
 
+    def getReferenceSets(self):
+        """
+        Returns the list of ReferenceSets in this backend
+        """
+        return list(self._referenceSetIdMap.values())
+
+    def getReference(self, id_):
+        """
+        Returns a reference with the given id_
+        """
+        return self.runGetRequest(self._referenceIdMap, id_)
+
+    def getReferenceSet(self, id_):
+        """
+        Returns a referenceSet with the given id_
+        """
+        return self.runGetRequest(self._referenceSetIdMap, id_)
+
+    def listReferenceBases(self, id_, requestArgs):
+        # parse arguments
+        try:
+            reference = self._referenceIdMap[id_]
+        except KeyError:
+            raise exceptions.ObjectWithIdNotFoundException(id_)
+        start = 0
+        end = datamodel.PysamDatamodelMixin.fastaMax
```

Comment:
`protocol.fromJsonDict` just assigns a value if there is an attribute name match; it doesn't check the type.  (Not sure if we want to change this.)

That is, we need to cast `requestArgs.start` to an int from a string ourselves.